core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackend.scala

```diff
@@ -88,7 +88,10 @@ private[spark] class MesosSchedulerBackend(
       throw new SparkException("Executor Spark home `spark.mesos.executor.home` is not set!")
     }
     val environment = Environment.newBuilder()
-    sc.conf.getOption("spark.executor.extraClassPath").foreach { cp =>
+
+    val userClassPath = sc.conf.getExecutorExtraClassPath()
+
+    userClassPath.foreach { cp =>
```

Comment:
nit: no real benefit to the tmp variable, you can keep it like before `sc.conf.getExecutorExtraClassPath().foreach { ...`